src/Button/DigitizeButton/DigitizeButton.jsx

```diff
@@ -552,12 +552,20 @@ class DigitizeButton extends React.Component {
       drawType,
       map,
       onDrawEnd,
-      onDrawStart
+      onDrawStart,
+      digitizeLayerName
     } = this.props;
 
     let geometryFunction;
     let type = drawType;
+    let digitizeLayer = MapUtil.getLayerByName(map, digitizeLayerName);
 
+    // this checks to see if the digitizeLayer  is in the map 
+    //and if not adds it from the state to the map
+    if (!digitizeLayer) {
+      map.addLayer(this.state.digitizeLayer);
```

Comment:
Just wondering why do we need this check here at all.. Actually the availibility of `digitizeLayer` will be ensured while `DigitizeButton` instantiation in `componentDidMount` method by calling `createDigitizeLayer()` function (compare (here)[https://github.com/terrestris/react-geo/blob/master/src/Button/DigitizeButton/DigitizeButton.jsx#L345] and (here)[https://github.com/terrestris/react-geo/blob/master/src/Button/DigitizeButton/DigitizeButton.jsx#L400].
So after this method is called, the `digitizeLayer` should be already written into state and be availiable via `this.state.digitizeLayer`.

It would be nice if you could recheck this again please..

Reply:
Ping @amir-ba.

Reply:
Well this is mainly for the case where the drawing Vector layer is removed from the map for any reason and the component has not changed. This causes the `digitizeLayer` to return null . and nothing is drawn. 
This is at least what I experienced.

Reply:
that is an error then. is the existance check enough at all? which removed the previousöy created layer?

Reply:
@marcjansen you are right I later realized the application bug and fixed it but I thought this extra check can't harm anyone and it keeps the consistency of the tool in case these cases happen. Anyhow @annarieger can decide if it is necessary at all to keep it in.

Reply:
Ok, thanks for explanation, I don't mind to leave this check in.

Please fix whitespace issues mentioned from other devs above and merge afterwards, thanks.